src/picongpu/include/particles/synchrotronPhotons/PhotonCreator.hpp

```diff
@@ -0,0 +1,289 @@
+     */
+    DINLINE unsigned int numNewParticles(FrameType& sourceFrame, int localIdx)
+    {
+        using namespace PMacc::algorithms;
+
+        PMACC_AUTO(particle, sourceFrame[localIdx]);
+
+        ValueType_E fieldE;
+        ValueType_B fieldB;
+
+        /* particle position, used for field-to-particle interpolation */
+        floatD_X pos = particle[position_];
+        const int particleCellIdx = particle[localCellIdx_];
+        /* multi-dim coordinate of the local cell inside the super cell */
+        DataSpace<TVec::dim> localCell(DataSpaceOperations<TVec::dim>::template map<TVec > (particleCellIdx));
+        /* interpolation of E- */
+        const fieldSolver::numericalCellType::traits::FieldPosition<FieldE> fieldPosE;
+        fieldE = Field2ParticleInterpolation()
+            (cachedE.shift(localCell).toCursor(), pos, fieldPosE());
+        /*                     and B-field on the particle position */
+        const fieldSolver::numericalCellType::traits::FieldPosition<FieldB> fieldPosB;
+        fieldB = Field2ParticleInterpolation()
+            (cachedB.shift(localCell).toCursor(), pos, fieldPosB());
+
+        const float3_X mom = particle[momentum_] / particle[weighting_];
+        const float_X mom2 = math::dot(mom, mom);
+        const float3_X mom_norm = mom * math::rsqrt(mom2);
+        const float_X mass = frame::getMass<FrameType>();
+        const float_X mc = mass * SPEED_OF_LIGHT;
+
+        const float_X gamma = math::sqrt(float_X(1.0) + mom2 / (mc*mc));
+        const float3_X vel = mom / (gamma * mass); // low accuracy?
+
+        const float3_X lorentz = fieldE + math::cross(vel, fieldB);
+        const float_X lorentz2 = math::dot(lorentz, lorentz);
+        const float_X fieldE_long = math::dot(mom_norm, fieldE);
+
+        const float_X H_eff = math::sqrt(lorentz2 - fieldE_long*fieldE_long);
+
+        const float_X charge = math::abs(frame::getCharge<FrameType>());
+        const float_X c = SPEED_OF_LIGHT;
+        // Schwinger limit, unit: V/m
+        const float_X E_S = mass*mass * c*c*c / (charge * HBAR);
+
+        const float_X chi = gamma * H_eff / E_S;
+
+        const float_X z = this->randomGen();
+        if(z == float_X(0.0) || z == float_X(1.0))
+            return 0;
+
+        const float_X x = emission_prob_modified(z, chi, gamma, mass, charge);
+
+        if(this->randomGen() < x)
+        {
+            this->photon_mom = mom_norm * z*z*z * mass*c * gamma * particle[weighting_];
+            return 1;
+        }
+
+        return 0;
+    }
+
+    /** Functor implementation.
+     *
+     * Called once for each single particle creation.
+     *
+     * \tparam Electron type of electron which creates the photon
+     * \tparam Photon type of photon that is created
+     */
+    template<typename Electron, typename Photon>
+    DINLINE void operator()(Electron& electron, Photon& photon) const
+    {
+        photon[multiMask_] = 1;
```

Comment:
please use something like this:

```C++
#include "particles/operations/Assign.hpp"
#include "particles/operations/Deselect.hpp"
...

photon[multiMask_] = 1;
photon[momentum_] = this->photon_mom;
electron[momentum_] -= this->photon_mom;
PMACC_AUTO( destPhoton, 
    deselect<
        boost::mpl::vector<
            multiMask,
            momentum        
        >(photon) );
particles::operations::assign( destPhoton, electron );
```

Than
- all attributes from electron were copied to phothon
- it is save whether electron has more attributes than phothon
- if the phothon has a attribute which is defined in electron this one is set to the user defined default value